src/cluster_legacy.c

```diff
@@ -4672,12 +4673,15 @@ int clusterGetFailedPrimaryRank(void) {
         clusterNode *node = dictGetVal(de);
 
         /* Skip nodes that do not need to participate in the rank. */
-        if (!nodeFailed(node) || !clusterNodeIsVotingPrimary(node) || node->num_replicas == 0) continue;
+        if (!(nodeFailed(node) || nodeTimedOut(node)) || !clusterNodeIsVotingPrimary(node) || node->num_replicas == 0) continue;
```

Comment:
It doesn't hurt to explain the conditional statement in code comment. Would be easier to understand why we choose to skip the failed/partially failed node.